pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -63,27 +69,32 @@ func (w *PodWatcher) computeChanges(podList []*Pod) ([]*Pod, error) {
 			continue
 		}
 
+		var newStaticPod, phaseChange bool
 		podEntity := PodUIDToEntityName(pod.Metadata.UID)
-		newStaticPod := false
-		_, found := w.lastSeen[podEntity]
+		prevSighting, found := w.lastSeen[podEntity]
+
 		// static pods are included specifically because they won't have any container
 		// as they're not updated in the pod list after creation
-		if isPodStatic(pod) == true && found == false {
-			newStaticPod = true
+		if found == false {
+			if isPodStatic(pod) == true {
+				newStaticPod = true
+			} else if prevSighting.phase != pod.Status.Phase {
+				phaseChange = true
+			}
 		}
 
-		// Refresh last pod seen time
-		w.lastSeen[podEntity] = now
+		// Refresh last pod seen time & phase
+		w.lastSeen[podEntity] = PodSighting{ts: now, phase: pod.Status.Phase}
 
 		// Detect new containers
 		newContainer := false
 		for _, container := range pod.Status.Containers {
-			if _, found := w.lastSeen[container.ID]; found == false {
+			if prevSighting, found := w.lastSeen[container.ID]; found == false || prevSighting.phase != pod.Status.Phase {
```

Comment:
As we already have `phaseChange` set, could we directly refresh every container of that pod without added logic?